File: java/Arrays and Strings/Anagram.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Anagram {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String str1 = in.nextLine();
        String str2 = in.nextLine();
        System.out.println(anagram(str1, str2));
    }

    public static boolean anagram(String str1, String str2) {
        int char_set[] = new int[256];

        for (int i = 0; i < str1.length(); i++)
            char_set[(int) str1.charAt(i)]++;

        for (int i = 0; i < str2.length(); i++) {
            if (char_set[(int) str2.charAt(i)] > 0)
                char_set[(int) str2.charAt(i)]--;
            else
                char_set[(int) str2.charAt(i)]++;
        }

        int sum = 0;
        for (int i = 0; i < 256; i++) sum += char_set[i];

        if (sum ==0) return true;

        return false;
    }
}

File: js/Arrays and Strings/anagram.js
'use strict';

function anagram(str1, str2) {
    var char_set = new Array(256);
    for (var i = 0; i < 256; i++) char_set[i] = 0;

    var val;

    for (var i = 0; i < str1.length; i++) {
        val = str1.charCodeAt(i);
        char_set[val]++;
    }

    for (var i = 0; i < str2.length; i++) {
        val = str2.charCodeAt(i);
        if (char_set[val] > 0)
            char_set[val]--;
        else
            char_set[val]++;
    }

    var sum = 0;
    for (var i = 0; i < 256; i++) sum += char_set[i];

    if (sum === 0) return true;

    return false;

}
